### PR TITLE
Reposition files routine

### PR DESCRIPTION
**Pull request by vkbo**
This PR just moves the reposition files routine from the scatter module to mod_units, so it can be used by any module. This is especially useful for collimation.

The routine has been cleaned up a bit, made general (also should work outside of C/R) and more comments added.

### CODE REVIEW

File: source/mod_units.f90 [hunk 1]
```diff
@@ -483,6 +483,100 @@ subroutine f_flush(unit)
...
+  logical isOpen, fErr
+  integer iError
+  integer j
+  character(len=mInputLn) aRecord
+
+  call f_requestUnit(fileName,fileUnit)
+  inquire(unit=fileUnit,opened=isOpen)
+  if(isOpen) then
+#ifdef CR
+    write(crlog,"(a,i0,a)") "UNITS> ERROR Failed while repsositioning '"//trim(fileName)//"', "//&
```
Comment (amereghe) [inline source/mod_units.f90]: repositioning

File: source/mod_units.f90 [hunk 2]
```diff
@@ -483,6 +483,100 @@ subroutine f_flush(unit)
...
+  integer,          intent(inout) :: fileUnit
+  integer,          intent(inout) :: filePos
+  integer,          intent(in)    :: newPos
+
+  logical isOpen, fErr
+  integer iError
+  integer j
+  character(len=mInputLn) aRecord
+
+  call f_requestUnit(fileName,fileUnit)
```
Comment (amereghe) [inline source/mod_units.f90]: shouldn't this call be under if statement? `f_requestUnit` initialises fileUnit to -1

File: source/mod_units.f90 [hunk 3]
```diff
@@ -483,6 +483,100 @@ subroutine f_flush(unit)
...
+    if(fErr) goto 10
+    filePos = 0
+    do j=1,newPos
+      read(fileUnit,"(a)",end=10,err=10,iostat=iError) aRecord
+      filePos = filePos + 1
+    end do
+    endfile(fileUnit,iostat=iError)
+    call f_close(fileUnit)
+
+    call f_open(unit=fileUnit,file=fileName,formatted=.true.,mode="w+",err=fErr,status="old")
```
Comment (amereghe) [inline source/mod_units.f90]: Maybe add a comment explaining why we need to end the file, close it and re-open it again

File: source/mod_units.f90 (hunk 2 above)
Comment (vkbo) [reply source/mod_units.f90]: What should the if-statement check?

Comment (amereghe) [reply source/mod_units.f90]: clarified by discussion - fine

**Comment by vkbo** [comment]
Noticed that the handling of `iostat` in `f_open` could return an uninitialised value, so fixed that as well.

**Pull request merged**